### PR TITLE
Fix incorrect wat in tests

### PR DESCRIPTION
**Pull request by tlively**
The new wat parser is much more strict than the legacy wat parser; the latter
accepts all sorts of things that the spec does not allow. To ease an eventual
transition to using the new wat parser by default, update the tests to use the
standard text format in many places where they previously did not. We do not yet
have a way to prevent new errors from being introduced into the test suite, but
at least there will now be many fewer errors when it comes time to make the
switch.

### CODE REVIEW

**Comment by tlively** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/WebAssembly/binaryen/6207?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `main`
  * **PR #6205** <a href="https://app.graphite.dev/github/pr/WebAssembly/binaryen/6205?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #6207** <a href="https://app.graphite.dev/github/pr/WebAssembly/binaryen/6207?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/WebAssembly/binaryen/6207?utm_source=stack-comment).

**Comment by tlively** [comment]
Oops, looks like this includes the commit for the previous PR in it. Please just review "[Fix incorrect wat in tests](https://github.com/WebAssembly/binaryen/pull/6207/commits/ded852919715434253fe341a6c0bac9dc446a448)"

File: test/passes/optimize-instructions_fuzz-exec.wast [hunk 1]
```diff
@@ -4,57 +4,57 @@
  (func $test32 (export "test32")
   (call $logf32
    (f32.add
-    (f32.const -nan:0xffff82)
+    (f32.const -nan:0x7fff82)
```
Comment (kripken) [inline test/passes/optimize-instructions_fuzz-exec.wast:7]: Why must these nans change?

Comment (tlively) [reply test/passes/optimize-instructions_fuzz-exec.wast:7]: The original NaNs here have invalid payloads. The spec for this is [here](https://webassembly.github.io/spec/core/text/values.html#floating-point).

The `fNMag` production `'nan:0x' n:hexnum => nan(n)` has a side condition `(1 <= n < 2^signif(N))` and `signif(32) = 23`, so the largest allowed NaN payload for a 32-bit float is `2^23 - 1 = 0x7FFFFF`.

Comment (kripken) [reply test/passes/optimize-instructions_fuzz-exec.wast:7]: Thanks, I didn't know that.

**Review by kripken — APPROVED** [review]

**Pull request merged**